src/main/java/teammates/ui/webapi/output/SessionResultsData.java

```diff
@@ -163,6 +164,8 @@ private static String removeAnonymousHash(String identifier) {
 
             for (FeedbackResponseAttributes response : responsesForRecipient) {
                 String giverName = removeAnonymousHash(bundle.getGiverNameForResponse(response));
+                String giverEmail = bundle.getDisplayableEmailGiver(response);
```

Comment:
I think `giverEmail` will never be null in this case, it would be `Anonymous Student xxx@Anonymous Student xxx.com` for anonymous response in this case. However this is not what we want right? Maybe follow the hint by checking `isGiverVisible` and `isRecipientVisible`?

Reply:
So we send a null from backend if the Giver/Recipient is not visible right?

Reply:
Of course or why we use `@Nullable`?